### DISCUSSION TITLE
Potentially excessive memory request in Vulnerability Scanner deployment

### DISCUSSION THREAD

**Discussion by conor-nsurely** [kubescape/kubescape#512] [General]
Hi everyone, just started using kubescape and loving it so far. Using `version v2.0.153`, installed following these instructions [these instructions](https://hub.armo.cloud/docs/installation-of-armo-in-cluster)

The vulnerability scanner deployment initially wasn't scheduled due to insufficient memory. The memory `request` in the deployment(armo-vuln-scan) is for `2500Mi`.

After starting the pod I ran some quick tests. Idle memory usage for the `armo-vuln-scan` pod is somewhere around 170Mi, with the highest value reached during a scan being 1187Mi for a few moments.

The cluster I have been testing on consists of 3 nodes, with ~40 pods running.

Do the memory requirements increase rapidly as the number of pods/nodes increases?  Perhaps a lower value for the request would be sufficient.

**Comment by dwertent** [comment]
Hi @conor-nsurely ,
I'm happy to hear that you are enjoying Kubescape, we are constantly working on developing the tool so it will be better for everyone to use.

The memory request is indeed a task we are currently investigating, I agree with you that the default is high but this is because we had some users with relatively large clusters. As I mentioned, we are investigating this matter.
We do not set the memory based on the size of the cluster, 
although it's not a bad idea at all. 
 
Thank you for your feedback, I encourage you to share any other feedback you have.
